### PR TITLE
[RUNTIME] Fix TVMContext for packed function

### PR DESCRIPTION
**Pull request by merrymercy**
Add missing definitions for using TVMContext in packed function

@tqchen Please review

### CODE REVIEW

**Comment by tqchen** [comment]
We intentionally try not to support TVMContext as a return value(argument value) in the python interface(possibly on a MAC). Seems that it breaks a case on a specific platform, let me spend sometime to confirm this. In the meanwhile, try to directly pass in arguments(device_id, device_type)

**Comment by merrymercy** [comment]
If it is not supported intentionally, then we should delete this line
https://github.com/dmlc/tvm/blob/d173e6374b87a70ea2cf4378a304bcb8fa9fa545/python/tvm/_ffi/_ctypes/function.py#L112-L114

**Comment by tqchen** [comment]
OK, the problem is that we cannot introduce Structure(TVMContext) as a member of ctypes.Union, which will cause some trouble in Mac and cause segfault.

I tried a bit and find the following alternative works(try to pack TVMContext into v_int64 field)

```python
def _context_to_int64(ctx):
    data = struct.pack("@ii", ctx.device_type, ctx.device_id)
    return struct.unpack("@q", data)[0]

def _context_from_tvm_value(value):
    data = struct.pack("@q", value.v_int64)
    arr = struct.unpack("@ii", data)
    return TVMContext(arr[0], arr[1])
```
Please update the PR to support this, and also add support for returning a TVMContext

File: tests/python/unittest/test_runtime_packed_func.py [hunk 1]
```diff
@@ -69,6 +69,13 @@ def myfunc(ss):
     x = tvm.convert(())
     tvm.convert(myfunc)(x)
 
+def test_ctx():
+    @tvm.register_func
+    def test_ctx_func(ctx):
+        assert ctx.device_type == TVMContext.STR2MASK['gpu']
```
Comment (tqchen) [inline tests/python/unittest/test_runtime_packed_func.py]: assert ctx == tvm.gpu(7)

**Comment by tqchen** [comment]
@merrymercy please followup on this issue

**Pull request closed without merging**